### ISSUE TITLE
New comment polluting compiled CSS output

### ISSUE THREAD

**Issue by dcwarwick** [carbon-design-system/carbon#9672]
### Package

@carbon/type

### Package version

10.35.0

### Description

As part of PR #9480, comments like the following have been added to some SCSS files:
![image](https://user-images.githubusercontent.com/6385315/133490790-2407e069-4779-4558-9886-d1bf089e6a4f.png)

The problem is that this style of comment, `/* ... */`, in sass means that the comment is preserved into the CSS output, meaning that the output CSS from our builds now contains many hundreds of copies of this comment and have become substantially larger as a result. These comments only apply at build/lint time and do not need to be preserved into the CSS output. They should be changed to `// ...` style comments, which are discarded during sass compile.

@tw15egan @joshblack FYI

### CodeSandbox example

no

### Steps to reproduce

n/a

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by joshblack** [comment]
@dcwarwick do you know if the `// ...` syntax is supported by `stylelinet` for things like `stylelint-disable`?

> meaning that the output CSS from our builds now contains many hundreds of copies of this comment and have become substantially larger as a result

These comments should be stripped out by a minifier if I understand correctly, is that not happening or is the CSS not being run through a minifier currently? Similarly, I believe sass has a `compressed` option that would omit comments as well.

**Comment by dcwarwick** [comment]
@joshblack So far as I know, `stylelint-disable` and similar can use any comment style. A quick search through the Carbon repo shows that you have quite a lot of `// stylelint-disable....` going on, which are presumably working. However, it also shows that you have a lot of other `/* stylelint-disable.... */` going on too, most of which presumably don't contribute to `carbon-components.css` but a few do -- converting those to the `// ....` format would make a further small reduction to the non-minified CSS size.

Yes, it is only the non-minified CSS which is affected. The minified versions will be unaffected. The non-minified `carbon-components.css` has grown by just over 21KB by having 472 copies of the `/* stylelint-disable number-max-precision */` comment in it.

**Comment by dcwarwick** [comment]
(PS we run various CSS tests by compiling the CSS expanded, not compressed, and that's how I noticed it. The tests and snapshots would be just as valid if they used the compressed form, but it's much less clear to see what has changed using the minified CSS which is why we use the expanded form, which is also quicker to run during the test cycle)

**Comment by joshblack** [comment]
@dcwarwick great point on the codebase, I'll definitely look into it and see 👀  thanks for pointing that out.

What are you seeing in terms of the impact of the size for uncompressed output? It seems like you're using it for tests, is it that the tests are now slower, or is some other factor coming into play? In general, size of uncompressed output is not something we generally consider a bug but if you're noticing something negative that is impacting a production use-case or a common use-case I'd love to know about it so we can make sure we consider it in our work 👍

**Comment by dcwarwick** [comment]
Agreed -- bloat to the non-minimised built CSS is unfortunate but not otherwise a major issue in itself. I think I was more concerned that these comments simply do not belong in the generated output, so it's a mistake, albeit an ultimately harmless one. I probably shouldn't have used the "bug" tag, just a dev issue.

Our tests showed up the issue, but once we've regenerated the snapshots they continue to pass -- although we do have one test that checks that a whole set of "definitions-only" SCSS generates no actual CSS output, and I have had to modify that test because although it still generates no actual CSS it does now generate some comments! I fixed that test by switching to use a minified CSS compile for the test, which does indeed slow the test down slightly. Not a big problem, though, and no ultimate impact on any production use cases.
